servers/rendering/renderer_rd/renderer_scene_render_rd.cpp

```diff
@@ -4077,7 +4092,23 @@ int RendererSceneRenderRD::get_max_directional_lights() const {
 }
 
 bool RendererSceneRenderRD::is_low_end() const {
-	return low_end;
+	// by default we switch this on this (may be ignored in some implementations)
+	return GLOBAL_GET("rendering/driver/rd_renderer/use_low_end_renderer");
+}
+
+bool RendererSceneRenderRD::gi_enabled() const {
+	// usable by default (unless low end = true)
+	return true;
+}
+
+bool RendererSceneRenderRD::clustered_enabled() const {
+	// used by default.
+	return true;
+}
+
+bool RendererSceneRenderRD::volumetric_enabled() const {
+	// usable by default (unless low end = true)
+	return true;
 }
```

Comment:
These are fishy. We should probably think about how this is going to work before merging. As I understand it, the values of gi, cluster, and volumetric should all be enabled on high end and disabled on low end. Is there a reason to check them separately?

Reply:
We're going to have a new subclass in which these need to be disabled, so there we will be overriding this and disabling them. 

This is the problem where `RendererSceneRenderRD` contains a lot of supporting code for other renderers that isn't used in all renderers.

Personally I'd like to remove those from `RendererSceneRenderRD`, move them into classes like we did with GI and SKY, and then include them in any subclass in which they are needed.

The problem is that they are incredibly intertwined, I spend two or 3 days just moving GI from `RendererSceneRenderRD` to `RenderFC` and it just turning into a mess. Discussing this with @reduz he told me to not even try and just add a way to disable the logic